sdk/core/azure-core/test/ut/request_activity_policy_test.cpp

```diff
@@ -204,6 +204,7 @@ TEST(RequestActivityPolicy, Basic)
       std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policies;
       // Add the request ID policy - this adds the x-ms-request-id attribute to the pipeline.
       policies.emplace_back(std::make_unique<RequestIdPolicy>());
+      policies.emplace_back(std::make_unique<TelemetryPolicy>("my-service-cpp", "1.0.0.beta-2"));
```

Comment:
It's test collateral, it doesn't need to be semver, it can just be a random string. 

What's important is that the logic which creates the user-agent header inserts this string value according to the guidelines. The fact that the guidelines imply that this is semantically versioned isn't significant to the test.